iceoryx_hoofs/test/moduletests/test_objectpool.cpp

```diff
@@ -570,53 +583,64 @@ TEST_F(ObjectPool_test, pointerToIndexConversion)
     // conversion does not care about valid data, index1 still corresponds to ptr1
     EXPECT_THAT(pool.pointerToIndex(ptr1), Eq(index1));
     EXPECT_THAT(pool.indexToPointer(index1), Eq(ptr1));
+}
 
-    // legal pointer checks
-    auto first = reinterpret_cast<Foo*>(pool.m_first);
-    auto last = reinterpret_cast<Foo*>(pool.m_last);
-    auto alignedPtr = reinterpret_cast<Foo*>(pool.m_first + sizeof(Foo));
+TEST_F(ObjectPool_test, pointerToIndexLegalPointerConversion)
+{
+    data = 0;
+    (void)poolExposed.construct(data);
```

Comment:
Please either use `EXPECT_THAT`/`ASSERT_THAT` or `IOX_DISCARD_RESULT` instead of the void cast

Reply:
done